envisage/plugin_manager.py

```diff
@@ -177,15 +173,17 @@ def stop_plugin(self, plugin=None, plugin_id=None):
     # The plugins that the manager manages!
     _plugins = List(IPlugin)
 
-    def __plugins_changed(self, trait_name, old, new):
+    @observe("_plugins")
+    def _update_application_on_all_plugins(self, event):
         """ Static trait change handler. """
+        old, new = event.old, event.new
+        self._update_application_on_plugins(old, new)
 
-        self._update_plugin_application(old, new)
-
-    def __plugins_items_changed(self, trait_name, old, new):
+    @observe("_plugins:items")
+    def _update_application_on_changed_plugins(self, event):
```

Comment:
nit: maybe `_update_application_on_plugins_contents`? 
I'm not sure thats really any clearer though.  Feel free to ignore

Reply:
i can't think of a better name but i don't think `_update_application_on_plugins_contents` is better 😅  sorry. im going to merge for now.